src/fun/qotd/advanced/component_handler.go.tmpl

```diff
@@ -0,0 +1,216 @@
+{{ define "MissingPerms" }}
+  {{ $yPerms := getTargetPermissionsIn 204255221017214977 .CID }}
+  {{ $missing := cslice }}
+  {{ range $k, $v := .AllPerms }}
+    {{ if and (in $.ReqPerms $v) (ne (bitwiseAnd $yPerms $v) $v) }}
+      {{ $missing = $missing.Append $k }}
+    {{ end }}
+  {{ end }}
+  {{ return joinStr ", " $missing }}
+{{ end }}
+{{ define "ClearCfg" }}
+  {{ $ = . }}
+  {{ $.Del "SuggestionsOpen" }}
+  {{ $.Del "MentionRole" }}
+  {{ $.Del "Sticky" }}
+  {{ $.Del "Publish" }}
+  {{ $.Del "UseThreads" }}
+  {{ $.Set "Xtras" cslice }}
+  {{ return $ }}
+{{ end }}
+
+{{ if eq .StrippedID "suggest" }}
+  {{ $placeholder := print "Ex: " (slice (execAdmin "topic") 2) }}
+  {{ if gt (len $placeholder) 100 }}
+    {{ $placeholder = slice $placeholder 0 101 }}
+  {{ end }}
+  {{ sendModal (sdict "title" "Suggest a question!" "custom_id" "qotd-submit" "fields" (cslice (sdict "label" "Question" "required" true "placeholder" $placeholder))) }}
+{{ else }}
+  {{ $cfg := dbGet 0 "qotd-config" }}
+  {{ with $cfg }}
+    {{ $cfg = .Value }}
+  {{ else }}
+    Sorry, please run `{{ .ServerPrefix }}qotd setup` again.
+    {{ return }}
+  {{ end }}
```

Comment:
Can this not be reworked to use a guard-clause style instead, therefore reducing the indentation of remaining file by one level?

Something like
```
  {{ sendModal ... }}
  {{ return }}
{{ end }}
{{ $cfg := ... }}
{{/* ... */}}
```

Reply:
Ah yup good point, thank you.